src/services/maci/service.ts

```diff
@@ -568,16 +605,75 @@ export class MaciService {
       )
       .then((tx) => tx.wait());
 
-    await this.storage.register({
+    await this.register({
       id: EContracts.VkRegistry,
       contract: vkRegistryContract,
-      args: [],
-      network: await this.getNetwork(),
     });
 
     return vkRegistryContract.getAddress();
   }
 
+  async registerExistingContract({ id, address, args }: IRegisterExistingContractArgs) {
+    let abi
+    switch(id) {
+      case EContracts.InitialVoiceCreditProxy:
+        abi = ConstantInitialVoiceCreditProxyFactory.abi;
+        break;
+      case EContracts.EASGatekeeper:
+        abi = EASGatekeeperFactory.abi;
+        break;
+      case EContracts.Verifier:
+        abi = VerifierFactory.abi;
+        break;
+      case EContracts.PoseidonT3:
+        abi = PoseidonT3Factory.abi;
+        break;
+      case EContracts.PoseidonT4:
+        abi = PoseidonT4Factory.abi;
+        break;
+      case EContracts.PoseidonT5:
+        abi = PoseidonT5Factory.abi;
+        break;
+      case EContracts.PoseidonT6:
+        abi = PoseidonT6Factory.abi;
+        break;
+      case EContracts.PollFactory:
+        abi = PollFactoryFactory.abi;
+        break;
+      case EContracts.MessageProcessorFactory:
+        abi = MessageProcessorFactoryFactory.abi;
+        break;
+      case EContracts.TallyFactory:
+        abi = TallyFactoryFactory.abi;
+        break;
+      case EContracts.MACI:
+        abi = MACIFactory.abi;
+        break;
+      case EContracts.VkRegistry:
+        abi = VkRegistryFactory.abi;
+        break;
+      default:
+        throw new Error("No such contract!");
```

Comment:
just wanna decide contract here? since if we put it outside, then we need to provide import contracts and provide abi in the hooks.
So the question is: where should we provide the abi? either in the frontend UI components, hooks, or here?

Reply:
You can move it to constant file in the same folder